dnf/cli/commands/repoquery.py

```diff
@@ -50,6 +50,13 @@
 provides, requires, obsoletes, conflicts, sourcerpm,
 description, summary, license, url, reason"""
 
+ALLOWED_QUERY_TAGS = ('name', 'arch', 'epoch', 'version', 'release',
+                      'reponame', 'repoid', 'from_repo', 'evr', 'debug_name',
+                      'source_name', 'source_debug_name', 'installtime',
+                      'buildtime', 'size', 'downloadsize', 'installsize',
+                      'provides', 'requires', 'obsoletes', 'conflicts',
+                      'sourcerpm', 'description', 'summary', 'license', 'url',
+                      'reason')
```

Comment:
Now the list is in two places - `ALLOWED_QUERY_TAGS` and `QUERY_TAGS`. Could one of them be somehow generated from the other?
What do you think?

Reply:
Definitely. Something like (

``` diff
diff --git a/dnf/cli/commands/repoquery.py b/dnf/cli/commands/repoquery.py
index 098ee836..83023aca 100644
--- a/dnf/cli/commands/repoquery.py
+++ b/dnf/cli/commands/repoquery.py
@@ -443,7 +443,7 @@ class RepoQueryCommand(commands.Command):
 
     def run(self):
         if self.opts.querytags:
-            print(QUERY_TAGS)
+            print("\n".join(ALLOWED_QUERY_TAGS))
             return
 
         self.cli._populate_update_security_filter(self.opts)
```

) would work. I just noticed that `QUERY_TAGS` was formatted in a rather specific way and thought it might better to keep it that way. Are you okay with the above solution?

Reply:
It prints almost 30 lines of output which seems too much to me. Maybe wrapping it using textwrap module? It creates almost identical output like original code:
```
print('\n'.join(textwrap.wrap(', '.join(ALLOWED_QUERY_TAGS))))
```
Or generate it the other way round? `ALLOWED_QUERY_TAGS = QUERY_TAGS.replace('\n', ' ').split(', ')`

Reply:
I think printing each on its own line is nice. It mirrors that behavior of `rpm --querytags` and makes it easy to `grep` for a field name. Otherwise, I think your first solution is better than generating it the other way around.

Reply:
OK then, let's go with "rpm compatible" solution then.

Reply:
See https://github.com/rpm-software-management/dnf/pull/1861/commits/eedd5ff1d0b2f568fef5abb9a83211b037d04251